### PR TITLE
added potato game starting stuff

### PR DESCRIPTION
**Pull request by pdlla**
😱

### CODE REVIEW

File: src/game/game.erl [hunk 1]
```diff
@@ -0,0 +1,44 @@
+-module(game).
+-behaviour(gen_server).
+
+
+%%-define(INTERVAL, 60000). % One minute
+-define(INTERVAL, 1000). % One minute
```
Comment (antonbobkov) [inline src/game/game.erl]: 1000 = 1 minute??

File: src/game/game.erl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+%% TODO state record
+
+start_link(Verifiers) ->
+  gen_server:start_link(?MODULE, Verifiers, []).
+
+init(Verifiers) ->
+  io:format("starting game with verifiers ~p~n",[Verifiers]),
+  {ok, _TRef} = timer:send_interval(?INTERVAL, timer_trigger),
```
Comment (antonbobkov) [inline src/game/game.erl]: Didn't know this was a thing :o 
How do you stop it? o:

File: src/potatoserver_sup.erl [hunk 3]
```diff
@@ -30,8 +31,27 @@ start_link() ->
...
 %% Before OTP 18 tuples must be used to specify a child. e.g.
 %% Child :: {Id,StartFunc,Restart,Shutdown,Type,Modules}
-init([]) ->
-    {ok, { {one_for_all, 0, 1}, []} }.
+init(Port) ->
+  %% TODO game sup
+  {ok, {{rest_for_one, 1, 5},[
+    %% start potato_udp, it will register itself under potato_udp (access via whereis(potatoudp))
+    #{id => potato_udp,
+      start => {potato_udp, start_link, [Port]},
```
[the remark this answers is not in the document]
Comment (antonbobkov) [reply src/potatoserver_sup.erl]: such antipattern 😿

File: src/game/potato_game_sup.erl [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+-include_lib("stdlib/include/assert.hrl").
+
+start_link() ->
+  %% this will register with name potato_game_sup I think?
+  supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+  io:format("starting potato_game_sup~n"),
+  {ok, {{simple_one_for_one, 1, 5},[
```
Comment (antonbobkov) [inline src/game/potato_game_sup.erl]: What's simple_one_for_one?

File: src/potatoserver_app.erl [hunk 5]
```diff
@@ -15,7 +15,26 @@
...
+    %% I guess you could have also registered it and grabbed it globally :\
+    {ok, PotatoChildSpec} = supervisor:get_childspec(EverythingSup, potato_game_sup),
+    {error,{already_started, PotatoSup}} = supervisor:start_child(EverythingSup, PotatoChildSpec),
+
+    %% TODO create Verifiers
+    Verifiers = [],
+
+    potato_game_sup:add_game(PotatoSup, Verifiers),
+
+    %% TODO other stuff 😱
```
Comment (antonbobkov) [inline src/potatoserver_app.erl]: 😱

**Comment by antonbobkov** [comment]
I should really read up on supervisors and gen_servers!

Looks really neat :satisfied:

File: src/game/game.erl (hunk 1 above)
Comment (pdlla) [reply src/game/game.erl]: close enough 🙀

File: src/game/game.erl (hunk 2 above)
Comment (pdlla) [reply src/game/game.erl]: I think you can do something like timer:stop(_TRef)

Comment (pdlla) [reply src/game/game.erl]: or kill the process that started it 😱

File: src/game/potato_game_sup.erl (hunk 4 above)
Comment (pdlla) [reply src/game/potato_game_sup.erl]: I think only 1 type of child allowed and you can add as many of that child. With the other types, the children are static and you can refer to them via their id

**Comment by pdlla** [comment]
supervisors behavior can get a little specific and it's really annoying to get refs to the children that you start within a supervisor. 

gen_server behavior on the other hand is awesome :D.

**Pull request merged**